CorsixTH/Lua/humanoid_actions/meander.lua

```diff
@@ -65,8 +65,17 @@ local function meander_action_start(action, humanoid)
       humanoid:setDynamicInfoText(_S.dynamic_info.staff.actions.wandering)
     end
   end
+
+  -- Handymen may have an assigned parcel, but they are free to visit any staff
+  -- room to rest
+  local meander_parcel
+  if humanoid.humanoid_class == "Handyman" and humanoid.parcelNr > 0 and
```

Comment:
Handymen are the only staff that can have an assigned parcel, so do we need to check for the Handyman class? If we decided to allow other staff to have assigned parcels then we wouldn't need this check then either.

Reply:
The `Staff` baseclass has the `parcelNr` variable, so all staff have that variable. I don't know what they do with it at all, so the `Handyman` check is mostly about avoiding accidents.

Also I didn't consider what is needed for other staff to work within a parcel, eg can you build a room on 2 parcels? Thus, it seemed that a different patch would be better to solve those problems.

Reply:
I'm happy with that.

For future reference, I don't know how often it can be done, but it is possible to build a room across multiple parcels. Campaign level 9 (Eastville Hospital) has a corridor between parcels 4 and 8 that is wide enough for a 4-tile room to be placed. I don't think it changes anything for this patch though.

Reply:
Changing isResting to a not in room condition would handle the room on parcel borders condition also.

Reply:
Can you elaborate on what you mean mugmuggy? If you mean checking if handyman is in a specific room type we did think about that. But it's not that best way iirc.

Reply:
I just mean 'not humanoid:getRoom()'. isResting is just a proxy for staff room, but if a handyman meanders in a room, which does happen from time to time, you may actually desire them to stay in the room, ie machine to be repaired is currently in use. Normally I don't think this occurs on purpose, it just happens, as most of the time there is a check if they are in a room, queues leave action followed by meander. I just can't find a reference for meander waiting for machine repair

Reply:
Mugmuggy, function changed. I assume that's what was wanted?

Reply:
Isn't room `0` the corridor? If so, it won't work:
``` lua
> if 0 then print"yes" end
yes
```
ie `0` is not `false`.

Reply:
Ah, nevermind, it returns `nil` if outside a room.

Reply:
Suggested vs wanted. On the times they do meander whilst waiting for the machine to become free, they should meander just fine anywhere in the room. As for finding and allocating work, I assume the use tiles and the objects itself determine who gets called to the room. Its already possible to have multiple handymen arrive if there are multiple calls to the one room, just it may look weird when a handyman doesn't repair when room is built on a border.